packages/react/src/utils/useAfterExitAnimation.tsx

```diff
@@ -8,13 +8,17 @@ import { useLatestRef } from './useLatestRef';
  * Useful for unmounting the component after animating out.
  */
 export function useAfterExitAnimation(parameters: useAfterExitAnimation.Parameters) {
-  const { open, animatedElementRef, onFinished: onFinishedParam } = parameters;
+  const { enabled = true, open, animatedElementRef, onFinished: onFinishedParam } = parameters;
 
   const onFinished = useEventCallback(onFinishedParam);
   const runOnceAnimationsFinish = useAnimationsFinished(animatedElementRef);
   const openRef = useLatestRef(open);
 
-  useEnhancedEffect(() => {
+  React.useEffect(() => {
```

Comment:
Change to regular `useEffect` as the timing lets Motion know the exit animation styles have been applied inside a single `requestAnimationFrame` when checking for `element.getAnimations()`